### PR TITLE
changed the metadata type

### LINKED ISSUE #12: Let the metadata field for the charge be map[string]string.
The `CreateChargeReq` request of the coinbase API has a `metadata` field that can be set. It would be better to make it flexible to accept a `map[string][string]`.

Currently the structure of the `metadata` field is 
```

type CreateChargeReq struct {
	Name        string               `json:"name"`
	Description string               `json:"description"`
	LocalPrice  CreateChargePrice    `json:"local_price"`
	PricingType enum.PricingType     `json:"pricing_type"`
	Metadata    CreateChargeMetadata `json:"metadata"`
	RedirectURL string               `json:"redirect_url"`
	CancelURL   string               `json:"cancel_url"`
}


type CreateChargeMetadata struct {
	CustomerID   string `json:"customer_id"`
	CustomerName string `json:"customer_name"`
}
```

**Describe the solution you'd like**
```
type CreateChargeReq struct {
	Name        string               `json:"name"`
	Description string               `json:"description"`
	LocalPrice  CreateChargePrice    `json:"local_price"`
	PricingType enum.PricingType     `json:"pricing_type"`
	Metadata   map[string][string] `json:"metadata"`
	RedirectURL string               `json:"redirect_url"`
	CancelURL   string               `json:"cancel_url"`
}
```

**Additional context**
Its a much better to add custom fields on the charge metadata object.

### PR DESCRIPTION
**Pull request by ohmpatel1997**
This Pr changes the metadata type of the create charge request.

closes #12

### CODE REVIEW

File: example/main.go [hunk 1]
```diff
@@ -37,9 +37,9 @@ func main() {
 			Currency: "USD",
 		},
 		PricingType: enum.PricingTypeFixedPrice,
-		Metadata: entity.CreateChargeMetadata{
-			CustomerID:   "id_1005",
-			CustomerName: "Satoshi Nakamoto",
+		Metadata: map[string]string{
+			"CustomerID":   "id_1005",
+			"CustomerName": "Satoshi Nakamoto",
```
Comment (rizalgowandy) [inline example/main.go]:
```suggestion
			"customer_id":   "id_1005",
			"customer_name": "Satoshi Nakamoto",
```

**Review by rizalgowandy — APPROVED** [review]

**Comment by rizalgowandy** [comment]
the ci failed, please fix it

**Comment by ohmpatel1997** [comment]
On it. fixing it.

**Comment by ohmpatel1997** [comment]
Fixed.

**Comment by ohmpatel1997** [comment]
> Fixed.

@rizalgowandy  can u please run the pipeline again.

**Comment by ohmpatel1997** [comment]
@rizalgowandy  I think its related to the github integration, everything else works fine.

**Pull request merged**